components/bastion/setup.ftl

```diff
@@ -0,0 +1,173 @@
+[#ftl]
+[#macro azure_bastion_arm_genplan_segment occurrence]
+  [@addDefaultGenerationPlan subsets=[ "template", "parameters"] /]
+[/#macro]
+
+[#macro azure_bastion_arm_setup_segment occurrence]
+
+  [#local core = occurrence.Core]
+  [#local solution = occurrence.Configuration.Solution]
+  [#local resources = occurrence.State.Resources]
+  [#local autoScaleConfig = solution.AutoScaling]
+
+  [#-- Network Lookups & Links --]
+  [#-- As full subnet name is vnet/subnet, retrieve both & format --]
+  [#local networkLink = getOccurrenceNetwork(occurrence).Link!{} ]
+  [#local networkLinkTarget = getLinkTarget(occurrence, networkLink, false) ]
+
+  [#if ! networkLinkTarget?has_content ]
+      [@fatal message="Network could not be found" context=networkLink /]
+      [#return]
+  [/#if]
+
+  [#local networkConfiguration = networkLinkTarget.Configuration.Solution]
+  [#local networkResources = networkLinkTarget.State.Resources ]
+  [#local networkVnetResource = networkResources["vnet"]]
+  [#local mgmtSubnetResource = networkResources["subnets"]["mgmt"]["subnet"]]
```

Comment:
The network shouldn't be fixed inside of the template and should be determined based on the tier the component is being deployed into

Reply:
See the network configuration in https://github.com/codeontap/gen3/blob/581df5b7796f21140b1739eff87794dcf81c2227/providers/aws/components/bastion/setup.ftl#L44 

The get getOccurrenceNetwork is the one which figures this out I'm pretty sure

Reply:
I believe this is the case of an unnecessary variable - networkConfiguration was not used anywhere. Same appears to be the case in the AWS provider.

Nonetheless I've re-arranged these vars - in bc54b00

Reply:
Cool, even better